pkg/comp-functions/functions/vshnnextcloud/collabora.go

```diff
@@ -0,0 +1,732 @@
+			Labels:    map[string]string{"app": comp.GetName() + "-collabora-code"},
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      comp.GetName() + "-collabora-code-sa",
+				Namespace: comp.GetInstanceNamespace(),
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			Kind:     "Role",
+			Name:     comp.GetName() + "-collabora-code-role",
+			APIGroup: "rbac.authorization.k8s.io",
+		},
+	}
+
+	return svc.SetDesiredKubeObject(roleBinding, comp.GetName()+"-collabora-code-role-binding", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createWatchOnlyObject(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	// Create a watch-only object to trigger a re-deploy
+	wo := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "collabora-code-tls",
+			Namespace: comp.GetInstanceNamespace(),
+		},
+	}
+
+	return svc.SetDesiredKubeObject(wo, comp.GetName()+watchOnlySecretObjectName, runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createSecretForOpenshiftRoute(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	secretFromWatchOnly := &corev1.Secret{}
+
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code-ingress-certificate",
+			Namespace: comp.GetInstanceNamespace(),
+			Labels:    map[string]string{"app": comp.GetName() + "-collabora-code"},
+		},
+		Data: map[string][]byte{},
+	}
+
+	err := svc.GetObservedKubeObject(secretFromWatchOnly, comp.GetName()+watchOnlySecretObjectName)
+	if err != nil {
+		svc.Log.Error(err, "Failed to get watch-only secret form collabora.createSecretForOpenshiftRoute()")
+		return err
+	}
+
+	secret.Data["tls.crt"] = secretFromWatchOnly.Data["ca.crt"]
+
+	return svc.SetDesiredKubeObject(secret, comp.GetName()+"collabora-code-ingress-certificate", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createInstallCollaboraConfigMap(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "collabora-install-script",
+			Namespace: comp.GetInstanceNamespace(),
+			Labels:    map[string]string{"app": comp.GetName() + "-collabora-code"},
+		},
+		Data: map[string]string{
+			"install-collabora.sh": installCollabora,
+		},
+	}
+
+	return svc.SetDesiredKubeObject(cm, comp.GetName()+"-collabora-install-script", runtime.KubeOptionAddLabels(labelMap))
+}
+
+func createInstallCollaboraJob(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+
+	// get hash from config, se job will be triggered if script changes
+
+	observedJob := &batchv1.Job{}
+	hash := md5.New()
+
+	_, _ = hash.Write([]byte(installCollabora))
+	checksum := hash.Sum(nil)
+
+	strChecksum := fmt.Sprintf("%x", checksum)
+	err := svc.GetObservedKubeObject(observedJob, comp.GetName()+"-collabora-install-job")
+	if err == nil {
+		observedHash, ok := observedJob.Labels["script"]
+		if ok && (observedHash != strChecksum) {
```

Comment:
Why all this checksum check? Isn't it enough to see if the object already exists? I feel the logic is overkill.

Reply:
there's a comment explaining the purpose of it:
```
	// get hash from config, se job will be triggered if script changes
```
whenever our script changes, for example thanks to our new release we will trigger new pod and run the logic

Reply:
Why should we re-run the script if collabora would be already installed? This is a script to install collabora, a one time script, what would happen if I re-run the script?

Reply:
it self detects if collabora is already installed and exit early if that's true